test/functional/cursor.test.js

```diff
@@ -405,74 +406,66 @@ describe('Cursor', function () {
 
             var cursor = collection.find().sort(['a', 1]);
-            test.deepEqual(['a', 1], cursor.sortValue);
-            finished();
+            test.deepEqual({ a: 1 }, cursor.sortValue);
+            finished(cursor);
 
             cursor = collection.find().sort('a', 1);
-            test.deepEqual([['a', 1]], cursor.sortValue);
-            finished();
+            test.deepEqual({ a: 1 }, cursor.sortValue);
+            finished(cursor);
 
             cursor = collection.find().sort('a', -1);
-            test.deepEqual([['a', -1]], cursor.sortValue);
-            finished();
+            test.deepEqual({ a: -1 }, cursor.sortValue);
+            finished(cursor);
 
             cursor = collection.find().sort('a', 'asc');
-            test.deepEqual([['a', 'asc']], cursor.sortValue);
-            finished();
-
-            cursor = collection.find().sort([
-              ['a', -1],
-              ['b', 1]
-            ]);
-            var entries = cursor.sortValue.entries();
-            test.deepEqual(['a', -1], entries.next().value);
-            test.deepEqual(['b', 1], entries.next().value);
-            finished();
+            test.deepEqual({ a: 1 }, cursor.sortValue);
+            finished(cursor);
 
             cursor = collection.find().sort('a', 1).sort('a', -1);
-            test.deepEqual([['a', -1]], cursor.sortValue);
-            finished();
+            test.deepEqual({ a: -1 }, cursor.sortValue);
+            finished(cursor);
 
+            cursor = collection.find();
             cursor.next(err => {
               expect(err).to.not.exist;
               try {
                 cursor.sort(['a']);
               } catch (err) {
                 test.equal('Cursor is closed', err.message);
-                finished();
+                finished(cursor);
               }
             });
 
-            collection
-              .find()
-              .sort('a', 25)
-              .next(err => {
-                test.equal(
-                  "Illegal sort clause, must be of the form [['field1', '(ascending|descending)'], ['field2', '(ascending|descending)']]",
-                  err.message
-                );
-                finished();
-              });
+            cursor = collection.find();
+            try {
+              cursor.sort('a', 25);
+            } catch (err) {
+              test.equal('Invalid sort direction: 25', err.message);
+            }
+            cursor.next(() => {
+              finished(cursor);
+            });
 
-            collection
-              .find()
-              .sort(25)
-              .next(err => {
-                test.equal(
-                  "Illegal sort clause, must be of the form [['field1', '(ascending|descending)'], ['field2', '(ascending|descending)']]",
-                  err.message
-                );
-                finished();
-              });
+            cursor = collection.find();
+            try {
+              cursor.sort(25);
```

Comment:
This now throws a sync error, which is a breaking change?

Reply:
I think this is fine, but we should probably add a `!` to the commit and make a note that sort direction is now being validated more strictly.